query-planner-js/src/buildPlan.ts

```diff
@@ -2452,19 +2561,19 @@ class FetchDependencyGraph {
     handledConditions: Conditions,
   ): {
     main: TProcessed,
-    unhandled: UnhandledGroups,
+    state: ProcessingState,
     deferredGroups: SetMultiMap<string, FetchGroup>,
   } {
     const conditions = updatedConditions(group.conditions(), handledConditions);
     const newHandledConditions = mergeConditions(conditions, handledConditions);
     const { children, deferredGroups } = this.extractChildrenAndDeferredDependencies(group);
     const processed = processor.onFetchGroup(group, newHandledConditions);
     if (children.length == 0) {
-      return { main: processor.onConditions(conditions, processed), unhandled: [], deferredGroups };
+      return { main: processor.onConditions(conditions, processed), state: ProcessingState.empty(), deferredGroups };
     }
 
-    const groupIsOnlyParentOfAllChildren = children.every(g => g.parents().length === 1);
-    if (groupIsOnlyParentOfAllChildren) {
+    const state = ProcessingState.forChildrenOfProcessedGroup(group, children);
+    if (state.next.length > 0) {
```

Comment:
Note that there's a logic change here.

Before this change, this `if` block executes only if all children have exactly one parent. After this change, this `if` block executes if at least one child has exactly one parent.

This might be fine/intentional, given that you've also swapped out `children` down below for `state.next`. Though in that case, you should update the comment below to say e.g. `ready children` instead of just `children`.